tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -1917,5 +1917,55 @@ public void RaisesAppDomainUnhandledExceptionEvent (ApplePlatform platform)
 				var output = ExecuteWithMagicWordAndAssert (appExecutable, env);
 			}
 		}
+
+		[Test]
+		[TestCase (ApplePlatform.iOS, "ios-arm64;")]
+		[TestCase (ApplePlatform.MacOSX, "osx-arm64;osx-x64")]
+		[TestCase (ApplePlatform.MacCatalyst, "maccatalyst-x64;")]
+		[TestCase (ApplePlatform.TVOS, "tvos-arm64;")]
+		public void PartialAOTExceptCoreLib (ApplePlatform platform, string runtimeIdentifiers)
+		{
+			var project = "MySimpleApp";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+			Configuration.AssertRuntimeIdentifiersAvailable (platform, runtimeIdentifiers);
+
+			var project_path = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out var appPath);
+			Clean (project_path);
+			var properties = GetDefaultProperties (runtimeIdentifiers);
+			properties ["MtouchInterpreter"] = "-all,System.Private.CoreLib.dll";
+
+			DotNet.AssertBuild (project_path, properties);
+
+			var dllPath = Path.Combine (appPath, "aot-instances.dll");
+			Assert.That (dllPath, Does.Not.Exist, "Dedup optimization shouldn't been enabled for partial AOT compilation");
```

Comment:
This will always pass on macOS and Mac Catalyst, because assemblies are in a subdirectory of the app bundle (Contents/MonoBundle/*.dll).

I think I would just list *all* the files in the app bundle, and assert that none of them are named 'aot-instances.dll'

Reply:
Thank you. I wasn't sure if there was the existing function for `.dll` search, so I added a new one.